lib/private/assistant-client.ts

```diff
@@ -27,4 +29,35 @@ export class AssistantClientImp implements AssistantClient {
 	async search(req: SearchRequest): Promise<SearchResponse> {
 		return this.http.post('/search', { ...req, assistantId: this.id });
 	}
+	async chat(req: ChatRequest): Promise<ChatResponse> {
+		const res = this.http.fetchAndStreamData('/chat', { ...req, assistantId: this.id });
+		const decoder = new TextDecoder();
+		let current = '';
+
+		for await (const chunk of res) {
+			const buffer = new TextEncoder().encode(chunk);
+			const decodedChunk = decoder.decode(buffer, { stream: true });
+			const lines = decodedChunk.split('\n');
+
+			for (const line of lines.filter((l) => l.length > 0)) {
+				if (line.startsWith('data: ')) {
+					current = line.slice(6);
```

Comment:
if you go over all lines before making the parsing, wouldn't you parse only the last message?